src/mimeparser.rs

```diff
@@ -1988,15 +2000,17 @@ fn is_attachment_disposition(mail: &mailparse::ParsedMail<'_>) -> bool {
             .any(|(key, _value)| key.starts_with("filename"))
 }
 
-fn is_valid_deltachat_vcard(mail: &mailparse::ParsedMail) -> bool {
-    let Ok(body) = &mail.get_body() else {
-        return false;
+/// Returns the 1st part of summary text (i.e. before the dash if any) for a valid DeltaChat vCard.
+pub(crate) fn get_vcard_summary(vcard: &[u8]) -> Option<String> {
+    let vcard = str::from_utf8(vcard).ok()?;
+    let contacts = deltachat_contact_tools::parse_vcard(vcard);
+    let [c] = &contacts[..] else {
+        return None;
     };
-    let contacts = deltachat_contact_tools::parse_vcard(body);
-    if let [c] = &contacts[..] {
-        return deltachat_contact_tools::may_be_valid_addr(&c.addr);
+    if !deltachat_contact_tools::may_be_valid_addr(&c.addr) {
+        return None;
     }
-    false
+    Some(c.display_name().to_string())
```

Comment:
Nit: This function could be moved to `message.rs`, I think it would fit there slightly better, but feel free to ignore